.github/workflows/build.yml

```diff
@@ -42,6 +42,83 @@ jobs:
             -Psonatype.username=${{ secrets.SONATYPE_USERNAME }}
             -Psonatype.password=${{ secrets.SONATYPE_PASSWORD }}
 
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+      - uses: gradle/wrapper-validation-action@v1
+      - name: Unit tests
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: testDebugUnitTest
+      - name: Upload failed unit test artifacts
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: unit-test-failures
+          path: '**/build/reports'
+
+  instrumentation-tests:
+    name: Instrumentation tests
+    runs-on: macOS-latest
+    timeout-minutes: 20
+    strategy:
+      fail-fast: false
+      matrix:
+        api-level: [31]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+      - uses: gradle/wrapper-validation-action@v1
+      - name: AVD cache
+        uses: actions/cache@v3
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-${{ matrix.api-level }}
+      - name: create AVD and generate snapshot for caching
+        if: steps.avd-cache.outputs.cache-hit != 'true'
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: ${{ matrix.api-level }}
+          arch: x86_64
+          force-avd-creation: false
+          emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: false
+          script: echo "Generated AVD snapshot for caching."
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+      - name: Instrumentation tests
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: ${{ matrix.api-level }}
+          arch: x86_64
+          force-avd-creation: false
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: true
+          script: |
+            adb logcat > logcat.out &
+            ./gradlew :sandbox:connectedCheck --stacktrace
```

Comment:
It seems to be significantly faster to specify the module rather than execute `connectedCheck`. Annoyingly without specifying the module it tries to build the androidTest tasks for all the modules even though they don't have any tests.

I was trying to find a way to potentially remove the connectedCheck task from those modules to avoid them being run, but I'm struggling to find the right API.

Any thoughts? It potentially adds 4-5 minutes of extra waiting.

Reply:
Merge with `build` job so you have much less to compile for `connectedCheck`.

> `--stacktrace`

Not sure why do you need it.

Reply:
good catch. I was using this before due to a problem. will remove it